src/main/java/seedu/address/model/task/Entry.java

```diff
@@ -184,4 +188,39 @@ public final String tagsString() {
             return buffer.substring(0, buffer.length() - DELIMITER.length());
         }
     }
+    
+    //@@author A0121501E
+    public int compareTo(Entry o) {
```

Comment:
Hmmm, it's a bit hard to explain.
In short, the compareTo function is not meant for this.
Remove the `compareTo` function and `implement Comparable<Entry>`.
Add a function called `compareShowDate` or something instead.